app/src/main/java/org/mozilla/fenix/settings/wallpaper/WallpaperSettingsFragment.kt

```diff
@@ -32,7 +28,7 @@ class WallpaperSettingsFragment : Fragment() {
             setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
             setContent {
                 FirefoxTheme {
-                    WallpaperSettings(wallpaperManager)
+                    WallpaperSettings(requireComponents.appStore)
```

Comment:
You don't want to pass the store here because it will always react to any AppStore changes. You will want to observe the selected wallpaper before here.